packages/router/src/__tests__/router.test.tsx

```diff
@@ -1191,6 +1191,31 @@ test('jump to new route, then go back', async () => {
   await waitFor(() => screen.getByText('Home Page'))
 })
 
+test('redirect replacing route', async () => {
+  const ListWithDefaultParamsPage = (props) => {
+    if (props['_limit']) {
+      return <h1>List Page</h1>
+    }
+    return <Redirect to="/list?_limit=10" options={{ replace: true }} />
+  }
+  const TestRouter = () => (
+    <Router>
+      <Route path="/" page={HomePage} name="home" />
+      <Route path="/list" page={ListWithDefaultParamsPage} name="list" />
+    </Router>
+  )
+  const screen = render(<TestRouter />)
+
+  // starts on home page
+  await waitFor(() => screen.getByText('Home Page'))
+
+  act(() => navigate(routes.list()))
```

Comment:
```suggestion
  // This will navigate to /list, which will then redirect to /list?_limit=10
  // which will render `<h1>List Page</h1>`
  act(() => navigate(routes.list()))
```